spec/features/user_updates_review_spec.rb

```diff
@@ -0,0 +1,71 @@
+require 'rails_helper'
+
+feature 'user updates review' do
+  # As a user
+  # I want to be able to update my own reviews
+  # because my original opinion was bad and I should feel bad
+  #
+  # Acceptance Criteria:
+  # [ ] User should be able to update any review from their profile page as
+  #     described below.
+  # [ ] When viewing my own review on a book's show page, there should be an
+  #     edit button next to that review only.
+  # [ ] The button leads to the review's show page, with pre-populated data.
+  # [ ] Upon successful form submission, A user is redirected to the book's show
+  #     page, with the message "Review updated successfully!"
+  # [ ] Upon unsuccessful form submission, the edit page is re-rendered with the
+  #     appropriate errors.
+
+  let!(:users_review) { FactoryGirl.create(:review) }
+  let!(:other_review) { FactoryGirl.create(:review, book: users_review.book) }
+
+  before(:each) { login_as(users_review.user, scope: :user) }
+
+  def expect_fields_autopopulated
+    expect(find_field('Title').value).to eq users_review.title
+    expect(find_field('Review').value).to eq users_review.body
+  end
+
+  def fill_in_form
```

Comment:
Eventually consider pulling out your helper methods into a separate module (in a different file)